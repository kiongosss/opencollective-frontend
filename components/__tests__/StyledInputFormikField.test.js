import React from 'react';
import { Formik } from 'formik';
import { z } from 'zod';

import { snapshotWithoutClassNames as snapshot } from '../../test/snapshot-helpers';

import { FormikZod } from '../FormikZod';
import StyledInputFormikField from '../StyledInputFormikField';

describe('StyledInputFormikField', () => {
  it('renders default options', () => {
    snapshot(<Formik>{() => <StyledInputFormikField name="myAttribute" />}</Formik>);
  });

  describe('With Zod schema', () => {
    it('automatically adds attributes from Zod schema', () => {
      const schema = z.object({ myAttribute: z.string().min(10).max(100) });
      snapshot(<FormikZod schema={schema}>{() => <StyledInputFormikField name="myAttribute" />}</FormikZod>);
    });

    it('automatically adds attributes from Zod schema on nested field', () => {
      const schema = z.object({ my: z.object({ nested: z.object({ attribute: z.string().min(10).max(100) }) }) });
      snapshot(<FormikZod schema={schema}>{() => <StyledInputFormikField name="my.nested.attribute" />}</FormikZod>);
    });

    it('automatically adds attributes from Zod schema on optional field', () => {
      const schema = z.object({ myAttribute: z.string().optional() });
      snapshot(<FormikZod schema={schema}>{() => <StyledInputFormikField name="myAttribute" />}</FormikZod>);
    });

    it('automatically adds attributes from Zod schema on nullable field', () => {
      const schema = z.object({ myAttribute: z.string().nullable() });
      snapshot(<FormikZod schema={schema}>{() => <StyledInputFormikField name="myAttribute" />}</FormikZod>);
    });

    it('automatically adds attributes from Zod schema on number field', () => {
      const schema = z.object({ myAttribute: z.number().min(10).max(100) });
      snapshot(<FormikZod schema={schema}>{() => <StyledInputFormikField name="myAttribute" />}</FormikZod>);
    });
  });
});
