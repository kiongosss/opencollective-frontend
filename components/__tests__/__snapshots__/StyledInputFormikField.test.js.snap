// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInputFormikField With Zod schema automatically adds attributes from Zod schema 1`] = `
<div>
  <div
    data-cy="InputField-myAttribute"
  >
    <div>
      <input
        fontSize="14px"
        id="input-myAttribute"
        maxLength={100}
        minLength={10}
        name="myAttribute"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`StyledInputFormikField With Zod schema automatically adds attributes from Zod schema on nested field 1`] = `
<div>
  <div
    data-cy="InputField-my.nested.attribute"
  >
    <div>
      <input
        fontSize="14px"
        id="input-my.nested.attribute"
        maxLength={100}
        minLength={10}
        name="my.nested.attribute"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`StyledInputFormikField With Zod schema automatically adds attributes from Zod schema on nullable field 1`] = `
<div>
  <div
    data-cy="InputField-myAttribute"
  >
    <div>
      <input
        fontSize="14px"
        id="input-myAttribute"
        name="myAttribute"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`StyledInputFormikField With Zod schema automatically adds attributes from Zod schema on number field 1`] = `
<div>
  <div
    data-cy="InputField-myAttribute"
  >
    <div>
      <input
        fontSize="14px"
        id="input-myAttribute"
        max={100}
        min={10}
        name="myAttribute"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="number"
      />
    </div>
  </div>
</div>
`;

exports[`StyledInputFormikField With Zod schema automatically adds attributes from Zod schema on optional field 1`] = `
<div>
  <div
    data-cy="InputField-myAttribute"
  >
    <div>
      <input
        fontSize="14px"
        id="input-myAttribute"
        name="myAttribute"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`StyledInputFormikField renders default options 1`] = `
<div>
  <div
    data-cy="InputField-myAttribute"
  >
    <div>
      <input
        fontSize="14px"
        id="input-myAttribute"
        name="myAttribute"
        onBlur={[Function]}
        onChange={[Function]}
      />
    </div>
  </div>
</div>
`;
